packages/gatsby/src/commands/develop.js

```diff
@@ -286,31 +290,12 @@ module.exports = async (program: any) => {
 
   let compiler
   await new Promise(resolve => {
```

Comment:
if we had a promise interface this would become

```
program.port = await detectPortInUseAndPrompt(port, rlInterface, newPort)

return startServer(program).then(([c, l]) => { compiler = c })
```